packages/loot-core/src/shared/util.ts

```diff
@@ -254,7 +254,12 @@ export let numberFormats = [
   { value: 'space-comma', label: '1 000,33', labelNoFraction: '1 000' },
 ];
 
-let numberFormat = {
+let numberFormat: {
+  value: string;
+  formatter: Intl.NumberFormat;
+  regex: RegExp;
+  separator?: string;
+} = {
```

Comment:
Why is this not failing the typechecker? Surely it should fail here. `value`, `formatter` and `regex` are set to `null` instead of the defined type.

Reply:
Indeed, this first pass with `strict: false` is to make the conversion easier. If we make it `true` now, it will make the process js -> ts much harder as we'll have to fix tons of things like type arguments for all functions, props for all components, anys and null checks, .... We will risk introducing bugs by trying to migrate to TS and fix TS complains. Moreover it will create many more conflicts with open PRs. At the end of the day the first TS pass is only slightly better than current JS code, but that is fine as will already catch few things and provide better IDE typing.

Overall, it is much safer to do one "transformation" at time, so I'll fix this case but my recommendation is to tackle strict as second pass (potentially enabling each strict flag individually first so you reduce the number of changes in review). I'm also happy to follow through the process until completion.

Reply:
Does `strictNullChecks` work without `strict` enabled?

Reply:
We can enable `strictNullChecks` individually, but it is `false` by default. As a test I've enabled it locally and it will flag 5 additional errors in the 2 files I've converted. However, its effectiveness is still poor given we lack types for arguments, so most of the types are "any".

Reply:
I think that’s OK! The other errors will appear as we transition more code over to TS and add type checking.

Reply:
The reason I'm not recommending to do it now is because to fix some of those errors we'll need to introduce runtime changes, and it easy to "fix" them with a change in behaviour that might cause a bug, hidden within several other type changes. I find it much easier to review PRs that only touch types rather than touching both types and runtime.